test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1613,27 +1752,28 @@ public synchronized String startDataOnlyNode(Settings settings) {
         return () -> submit.get();
     }
 
+
     /**
      * Starts multiple nodes in an async manner and returns future with its name.
      */
     public synchronized Async<List<String>> startNodesAsync(final int numNodes) {
-        return startNodesAsync(numNodes, Settings.EMPTY, Version.CURRENT);
+        return startNodesAsync(numNodes, Settings.EMPTY);
     }
 
     /**
      * Starts multiple nodes in an async manner with the given settings and returns future with its name.
      */
-    public synchronized Async<List<String>> startNodesAsync(final int numNodes, final Settings settings) {
-        return startNodesAsync(numNodes, settings, Version.CURRENT);
-    }
-
-    /**
-     * Starts multiple nodes in an async manner with the given settings and version and returns future with its name.
-     */
-    public synchronized Async<List<String>> startNodesAsync(final int numNodes, final Settings settings, final Version version) {
+    public synchronized Async<List<String>> startNodesAsync(final int numNodes, final Settings settings)  {
+        final int defaultMinMasterNodes;
+        if (autoManageMinMasterNodes) {
+           int mastersDelta = Node.NODE_MASTER_SETTING.get(settings) ? numNodes : 0;
+            defaultMinMasterNodes = updateMinMasterNodes(getMasterNodesCount() + mastersDelta);
+        } else {
+            defaultMinMasterNodes = -1;
+        }
```

Comment:
this code block is very similar to something I already saw above, shall we factor it out?